app/resources/api/v2/purpose_resource.rb

```diff
@@ -2,40 +2,73 @@
 
 module Api
   module V2
-    # @todo This documentation does not yet include a detailed description of what this resource represents.
-    # @todo This documentation does not yet include detailed descriptions for relationships, attributes and filters.
-    # @todo This documentation does not yet include any example usage of the API via cURL or similar.
+    # Provides a JSON:API representation of {Purpose}.
+    #
+    # A Purpose defines the intended function of a labware or sample within the system.
+    # While it was historically tied to PlatePurpose, it now applies to other labware like Tubes.
     #
     # @note Access this resource via the `/api/v2/purposes/` endpoint.
     #
-    # Provides a JSON:API representation of {Purpose}.
+    # @example Fetching all purposes
+    #   GET /api/v2/purposes
+    #
+    # @example Fetching a purpose by ID
+    #   GET /api/v2/purposes/{id}
     #
-    # For more information about JSON:API see the [JSON:API Specifications](https://jsonapi.org/format/)
-    # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
-    # of the JSON:API standard.
+    # @note the below example is currently broken, as `target_type` is a required attribute in the
+    #   model and `lifestan` is not.
```

Comment:
```suggestion
    #   model and `lifespan` is not.
```

Reply:
Should we add a @todo to fix this broken endpoint / example?

Reply:
Yes, I have added it as one of the many “Possible Bugs” [y24-478-todo.md](https://github.com/sanger/sequencescape/blob/d56a60b47acc8cd10f9dbcb8c49d050ea00dd3ad/app/resources/api/v2/y24-478-learnings.md) here. 

After the code review I was going ask the team what to do with all these “findings” !?